mopidy_musicbox_webclient/static/js/images.js

```diff
@@ -61,47 +64,92 @@
         },
 
         getCoverFromLastFm: function (track, images, size) {
-            var defUrl = 'images/default_cover.png'
-            $(images).attr('src', defUrl)
+            $(images).attr('src', this.DEFAULT_ALBUM_URL)
             if (!track || !(track.album || track.artists)) {
                 return
             }
-            var albumname = (track.album && track.album.name) ? track.album.name : ''
-            var artistname = ''
-            if (track.album && track.album.artists && track.album.artists.length > 0) {
+
+            var musicBrainzID = track.album.musicbrainz_id
+            var albumName = track.album.name
+            var artistName = this.getArtistInfo(track).name
+
+            var params = {}
+            if (musicBrainzID) {
+                params.mbid = musicBrainzID
+            }
+            if (artistName && albumName) {
+                params.artist = artistName
+                params.album = albumName
+            }
+            if ((params.artist && params.album) || params.mbid) {
+                this.lastFM.album.getInfo(params, {success: function (data) {
+                    for (var i = 0; i < data.album.image.length; i++) {
+                        if (data.album.image[i].size === size) {
+                            $(images).attr('src', data.album.image[i]['#text'] || this.DEFAULT_ALBUM_URL)
+                            break
+                        }
+                    }
+                }, error: function (code, message) {
+                    console.error('Error retrieving album info from last.fm', code, message)
+                }})
+            } else {
+                console.warn('Not enough tag information available to retrieve album cover from last.fm, using default')
+            }
+        },
+
+        getArtistInfo: function (track) {
+            var artistName = ''
+            var musicBrainzID = ''
+
+            if (track && track.album && track.album.artists && track.album.artists.length > 0) {
                 // First look for the artist in the album
```

Comment:
This is easy enough to change and I agree using track artist before album artist will probably make sense for most situations.